plugins/field-colour-hsv-sliders/src/field_colour_hsv_sliders.ts

```diff
@@ -0,0 +1,639 @@
+    return this;
+  }
+
+  /**
+   * Updates the properties of this instance to represent the same colour as the
+   * provided HsvColour but in the sRGB colour space.
+   * @param hsv An HSV representation of a colour to copy.
+   * @return This instance after updating it.
+   */
+  loadFromHsv(hsv: HsvColour): RgbColour {
+    const hue: number = (hsv.h - Math.floor(hsv.h)) * 6;
+    this.r = hsv.v * (1 - hsv.s * Math.max(0, Math.min(1,
+        2 - Math.abs(hue - 3))));
+    this.g = hsv.v * (1 - hsv.s * Math.max(0, Math.min(1,
+        Math.abs(hue - 2) - 1)));
+    this.b = hsv.v * (1 - hsv.s * Math.max(0, Math.min(1,
+        Math.abs(hue - 4) - 1)));
+    return this;
+  }
+}
+
+/**
+ * A structure with three properties h, s, and v, representing the hue,
+ * saturation, and brightness in a colour. All three properties range from 0
+ * to 1.
+ */
+class HsvColour {
+  /** The hue of the colour, ranging from 0 to 1. */
+  h: number;
+
+  /** The saturation of the colour, ranging from 0 to 1. */
+  s: number;
+
+  /** The brightness of the colour, ranging from 0 to 1. */
+  v: number;
+
+  /**
+   * The HsvColour constructor.
+   * @param h The initial hue of the colour. Defaults to 0.
+   * @param s The initial amount of saturation. Defaults to 0.
+   * @param v The initial amount of brightness. Defaults to 0.
+   */
+  constructor(h = 0, s = 0, v = 0) {
+    this.h = h;
+    this.s = s;
+    this.v = v;
+  }
+
+  /**
+   * Updates the properties of this instance to represent the same colour as the
+   * provided RgbColour but in the HSV colour space.
+   * @param rgb An RGB representation of a colour to copy.
+   * @return This instance after updating it.
+   */
+  loadFromRgb(rgb: RgbColour): HsvColour {
+    const max: number = Math.max(Math.max(rgb.r, rgb.g), rgb.b);
+    const min: number = Math.min(Math.min(rgb.r, rgb.g), rgb.b);
+    this.v = max;
+    if (min == max) {
+      this.h = 0;
+      this.s = 0;
+      return this;
+    }
+
+    const delta: number = (max - min);
+    this.s = delta / max;
+
+    let hue: number;
+    if (rgb.r == max) {
+      hue = (rgb.g - rgb.b) / delta;
+    } else if (rgb.g == max) {
+      hue = 2 + ((rgb.b - rgb.r) / delta);
+    } else {
+      hue = 4 + ((rgb.r - rgb.g) / delta);
+    }
+    hue /= 6;
+    this.h = hue - Math.floor(hue);
+    return this;
+  }
+
+  /**
+   * Updates the properties of this instance to copy the provided HsvColour.
+   * @param other An HSV representation of a colour to copy.
+   * @return This instance after updating it.
+   */
+  copy(other: HsvColour): HsvColour {
+    this.h = other.h;
+    this.s = other.s;
+    this.v = other.v;
+    return this;
+  }
+}
+
+/**
+ * Class for a colour input field that displays HSV slider widgets when clicked.
+ */
+export class FieldColourHsvSliders extends Blockly.FieldColour {
+  /* eslint-disable @typescript-eslint/naming-convention */
+  /** The maximum value of the hue slider range. */
```

Comment:
IIRC styleguide guidance is to omit types for simple direct assignments

Reply:
Done.